tensorflow/contrib/tensorrt/convert/convert_nodes_test.cc

```diff
@@ -1964,6 +1964,48 @@ TEST_F(OpConverterTest, ConvertRelu6) {
   }
 }
 
+TEST_F(OpConverterTest, ConvertSquare) {
+  {
+    // Input list is empty, should fail.
+    NodeDef node_def = MakeNodeDef("my_square", "Square", {});
+    RunValidationAndConversion(
+        node_def, error::INVALID_ARGUMENT,
+        "Square expects one input, at my_square");
+  }
+
+  // Get the NodeDef for Square.
+  Scope s = Scope::NewRootScope();
+  auto input = ops::Placeholder(s.WithOpName("input"), DT_FLOAT);
+  auto square = ops::Square(s.WithOpName("my_square"), input);
+  const NodeDef& node_def = square.operation.node()->def();
+
+  {
+    // Input is weights, should fail.
+    Reset();
+    AddTestWeights<int32>("input", {1, 2, 3}, {1, 2, 3, 4, -5, 6});
+    RunValidationAndConversion(node_def, error::UNIMPLEMENTED,
+        "Square is only implemented for tensors, at my_square");
+  }
+  {
+    // Input is tensor, Ok.
+    Reset();
+    AddTestTensor("input", {1, 2, 3});
```

Comment:
Could you test other data types like DT_HALF? To do that we can add a template function similar `TestConvertBiasAdd`.